account_cutoff_accrual_picking/models/account_cutoff.py

```diff
@@ -0,0 +1,268 @@
+                _("Error: Missing %s account on product '%s' or on "
+                    "related product category.") % (
+                        _(map_type), line.product_id.name))
+        return account
+
+    def _prepare_line(self, line):
+        """
+        Calculate accrued expense using purchase.order.line
+        or accrued revenu using sale.order.line
+        """
+        assert self.type in ('accrued_expense', 'accrued_revenue'),\
+            "The field 'type' has a wrong value"
+
+        partner = line.order_id.partner_id
+        fpos = partner.property_account_position_id
+        account_id = self._get_account(line, self.type, fpos).id
+        accrual_account_id = self._get_account_mapping().get(
+            account_id, account_id)
+
+        if self.type == 'accrued_expense':
+            # Processing purchase order line
+            analytic_account_id = line.account_analytic_id.id
+            price_unit = line.price_unit
+            taxes = line.taxes_id
+            taxes = fpos.map_tax(taxes)
+            tax_account_field_name = 'account_accrued_expense_id'
+            tax_account_field_label = 'Accrued Expense Tax Account'
+            quantity = line.qty_received - line.qty_invoiced
+
+        elif self.type == 'accrued_revenue':
+            # Processing sale order line
+            analytic_account_id = line.order_id.project_id.id or False
+            price_unit = line.price_unit * (1 - (line.discount or 0.0) / 100.0)
+            taxes = line.tax_id
+            taxes = fpos.map_tax(taxes)
+            tax_account_field_name = 'account_accrued_revenue_id'
+            tax_account_field_label = 'Accrued Revenue Tax Account'
+            quantity = line.qty_to_invoice
+
+        # Processing the taxes
+        company_currency = self.company_id.currency_id
+        currency = line.currency_id
+        tax_line_ids = []
+        tax_res = taxes.compute_all(price_unit, currency, quantity,
+                                    line.product_id, partner)
+        amount = tax_res['total_excluded']
+        if self.type == 'accrued_expense':
+            amount = amount * -1
+        for tax_line in tax_res['taxes']:
+            tax_read = self.env['account.tax'].browse(tax_line['id'])
+            tax_accrual_account_id = tax_read[tax_account_field_name]
+            if not tax_accrual_account_id:
+                raise exceptions.UserError(
+                    _("Error: Missing '%s' on tax '%s'.")
+                    % (tax_account_field_label, tax_read['name']))
+            else:
+                tax_accrual_account_id = tax_accrual_account_id[0]
+            if self.type == 'accrued_expense':
+                tax_line['amount'] = tax_line['amount'] * -1
+            if company_currency != currency:
+                currency_at_date = currency.with_context(date=self.cutoff_date)
+                tax_accrual_amount = currency_at_date.compute(
+                    tax_line['amount'], company_currency.id)
```

Comment:
change `company_currency.id` to `company_currency`

Reply:
Where is that compute method on currency defined? Can't find it :(

Reply:
It is here https://github.com/odoo/odoo/blob/3bd2bb7aa1e3448cd946cbe86533051e139b6258/odoo/addons/base/res/res_currency.py#L149
Is fix looks good to me, I asked him to propose the change

Reply:
`odoo/addons/base/res/res_currency.py`

Reply:
please review pr https://github.com/TDu/account-closing/pull/4

Reply:
Thanks merged.